neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/Neo4jDialect.java

```diff
@@ -6,592 +6,17 @@
-import org.hibernate.ogm.datastore.neo4j.embedded.dialect.impl.EmbeddedNeo4jTupleSnapshot;
-import org.hibernate.ogm.datastore.neo4j.embedded.dialect.impl.EmbeddedNeo4jNodesTupleIterator;
 import org.hibernate.ogm.datastore.neo4j.embedded.impl.EmbeddedNeo4jDatastoreProvider;
-import org.hibernate.ogm.datastore.neo4j.logging.impl.GraphLogger;
-import org.hibernate.ogm.datastore.neo4j.logging.impl.Log;
-import org.hibernate.ogm.datastore.neo4j.logging.impl.LoggerFactory;
-import org.hibernate.ogm.dialect.query.spi.BackendQuery;
-import org.hibernate.ogm.dialect.query.spi.ClosableIterator;
-import org.hibernate.ogm.dialect.query.spi.QueryParameters;
-import org.hibernate.ogm.dialect.spi.AssociationContext;
-import org.hibernate.ogm.dialect.spi.ModelConsumer;
-import org.hibernate.ogm.dialect.spi.NextValueRequest;
-import org.hibernate.ogm.dialect.spi.TupleAlreadyExistsException;
-import org.hibernate.ogm.dialect.spi.TupleContext;
-import org.hibernate.ogm.model.key.spi.AssociatedEntityKeyMetadata;
-import org.hibernate.ogm.model.key.spi.AssociationKey;
-import org.hibernate.ogm.model.key.spi.AssociationKeyMetadata;
-import org.hibernate.ogm.model.key.spi.EntityKey;
-import org.hibernate.ogm.model.key.spi.EntityKeyMetadata;
-import org.hibernate.ogm.model.key.spi.RowKey;
-import org.hibernate.ogm.model.spi.Association;
-import org.hibernate.ogm.model.spi.AssociationOperation;
-import org.hibernate.ogm.model.spi.EntityMetadataInformation;
-import org.hibernate.ogm.model.spi.Tuple;
-import org.hibernate.ogm.model.spi.TupleOperation;
-import org.hibernate.ogm.persister.impl.OgmCollectionPersister;
-import org.hibernate.ogm.persister.impl.OgmEntityPersister;
-import org.hibernate.persister.collection.CollectionPersister;
-import org.hibernate.persister.entity.EntityPersister;
-import org.neo4j.graphdb.ConstraintViolationException;
-import org.neo4j.graphdb.Direction;
-import org.neo4j.graphdb.GraphDatabaseService;
-import org.neo4j.graphdb.Node;
-import org.neo4j.graphdb.QueryExecutionException;
-import org.neo4j.graphdb.Relationship;
-import org.neo4j.graphdb.ResourceIterator;
-import org.neo4j.graphdb.Result;
-import org.neo4j.kernel.api.exceptions.schema.UniquePropertyConstraintViolationKernelException;
 
 /**
- * Abstracts Hibernate OGM from Neo4j.
- * <p>
- * A {@link Tuple} is saved as a {@link Node} where the columns are converted into properties of the node.<br>
- * An {@link Association} is converted into a {@link Relationship} identified by the {@link AssociationKey} and the
- * {@link RowKey}. The type of the relationship is the value returned by
- * {@link AssociationKeyMetadata#getCollectionRole()}.
- * <p>
- * If the value of a property is set to null the property will be removed (Neo4j does not allow to store null values).
- *
- * @author Davide D'Alto &lt;davide@hibernate.org&gt;
+ * @deprecated use {@link EmbeddedNeo4jDialect}. We might decide to remove this class in the future or change its
```

Comment:
I'll wait to deprecate it and leave it there for now, although it duplicates the Embedded dialect